ROADMAP.md

```diff
@@ -20,18 +20,60 @@ Create radio drivers on raspberry pi and desktop ends which provide a programmab
+
+### PoC - Define and implement block-ship re-transmissions
+
+The initial implementation of the block-ship protocol is a very simple one-way stream of DAG blocks. Define a mechanism for detecting missing blocks, requesting transmission of specific blocks, and responding to those requests. Implement in block-ship transmitter and receiver.
+
+### MVP v0.3 - Demonstrate block-ship retransmit over radio
+
+Establish a block-ship session over the dev radio link at a speed which ensures minor packet loss and tune the re-transmission mechanism to overcome the packet loss.
+
+### PoC - Define and implement block-ship storage mechanism
+
+Define and implement (or borrow) a method for chunking files into DAG blocks and storing said blocks in persistent storage. Chunked files should be retrievable by CID.
+
+### PoC - Define and implement block-ship CID request
+
+Define and implement a mechanism for block-ship to transmit & receive a CID request, and to respond to CID request with appropriate previously stored blocks.
+
+### MVP v0.4 - Demonstrate block-ship request by CID
+
+The raspberry pi will run a service which generates an in-memory DAG representation of a file, and can respond to a CID request for that file (CID will be shared ahead of time). The ground station sends a CID request over the radio to the raspberry pi, and the raspberry pi responds with the corresponding block-ship stream. The ground station receives the block-ship stream, reassembles, and verifies the payload. Radio transmission should be tuned for minor packet loss and protocol should reasonably overcome it.
+
+### PoC - Define and implement block-ship CID advertisements
+
+Define and implement a mechanism for block-ship to advertise which CIDs are available for request/transmission.
+
+### Poc - Define and implement basic spacecraft application API
+
+Define an interface and API by which other spacecraft applications/services can interact with the on-board IPFS instance to perform actions such as storing/chunking a file and requesting transmission of a file.
+
+### MVP v0.5 - Basic application API and CID advertisements
+
+An application API is defined and implemented for the raspberry pi which allows on-board applications to request that a file is stored in local IPFS. This includes the ability to mantain a list of CIDs stored on-board and to advertise that list to the ground. The ground station uses this advertised list to request a CID it does not currently have, and the raspberry pi transmits.
+
+### PoC - Initial communications API
+
+Define and document current method of integrating block-ship pieces into satellite and ground communications layers. Extract out UDP packet stuffing into abstract interface to allow for other packet formats in other use cases.
+
```

Comment:
Should we have another POC that uses different satellite profiles (earth observing, communications, navigation, weather etc.) to drive evaluation and metrics.

Reply:
Added a POC at the bottom to catch this, we can move up to an earlier milestone if needed.